dandi/models.py

```diff
@@ -600,12 +700,19 @@ def check_data(cls, values):
     assetsSummary: TempOptional[AssetsSummary] = Field(readOnly=True, nskey="dandi")
 
     # From server (requested by users even for drafts)
-    manifestLocation: TempOptional[List[AnyUrl]] = Field(readOnly=True, nskey="dandi")
+    manifestLocation: TempOptional[List[HttpUrl]] = Field(readOnly=True, nskey="dandi")
 
     # On publish
     version: TempOptional[str] = Field(readOnly=True, nskey="schema")
-    doi: Optional[Union[AnyUrl, str]] = Field(
-        None, title="DOI", readOnly=True, nskey="dandi"
+    doi: Optional[str] = Field(
+        None, title="DOI", readOnly=True, pattern="^10\.[A-Za-z0-9.\/-]+", nskey="dandi"
```

Comment:
`\.` is not a valid escape sequence.  Either the string should be made raw (with an `r` prefix) or another backslash should be added.  Without this, certain versions of Python produce a syntax error.